### ISSUE TITLE
SKR has a small bug

### ISSUE THREAD

**Issue by codecodebear** [RUC-NLPIR/FlashRAG#70]
Hi,

I get a bug when running ```SKR```. 

I have set ```model_path``` to ```/workspace/retrievers/sup-simcse-bert-base-uncased``` , and used the default ```training_data_path```.

The error message is below:
```
Retrieval process: 100%|██████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  3.82it/s]
Processed prompts: 100%|██████| 44/44 [00:00<00:00, 70.05it/s, est. speed input: 55448.66 toks/s, output: 562.07 toks/s]
Processed prompts: 100%|█████████| 6/6 [00:00<00:00, 61.95it/s, est. speed input: 3151.41 toks/s, output: 340.91 toks/s]
[rank0]: Traceback (most recent call last):
[rank0]:   File "/workspace/FlashRAG/examples/methods/run_exp.py", line 562, in <module>
[rank0]:     func(args)
[rank0]:   File "/workspace/FlashRAG/examples/methods/run_exp.py", line 337, in skr
[rank0]:     result = pipeline.run(test_data)
[rank0]:   File "/workspace/FlashRAG/flashrag/pipeline/pipeline.py", line 173, in run
[rank0]:     dataset = merge_dataset(dataset_split, judge_result)
[rank0]:   File "/workspace/FlashRAG/flashrag/dataset/utils.py", line 40, in merge_dataset
[rank0]:     final_dataset = Dataset(
[rank0]:   File "/workspace/FlashRAG/flashrag/dataset/dataset.py", line 77, in __init__
[rank0]:     self.dataset_name = config["dataset_name"]
[rank0]:   File "/workspace/FlashRAG/flashrag/dataset/dataset.py", line 150, in __getitem__
[rank0]:     return self.data[index]
[rank0]: TypeError: list indices must be integers or slices, not str
```
Can you take a look? Thank you!

**Comment by ignorejjj** [comment]
Sorry, this was an issue caused by the morning update, I have fixed it (commit 68c80db88094dc96789cb27405914fae8e3656c7).

**Comment by codecodebear** [comment]
Hi, thank you for the fix!
It outputs the metrics correctly, but there is one tiny issue that generates an error message.
```
Retrieval process: 100%|████████████████████████████████████████████████████| 12/12 [00:02<00:00,  4.96it/s]
Processed prompts: 100%|█| 2831/2831 [00:27<00:00, 103.11it/s, est. speed input: 82793.73 toks/s, output: 10
Processed prompts: 100%|█| 779/779 [00:01<00:00, 620.93it/s, est. speed input: 31868.70 toks/s, output: 4095
{'em': 0.30249307479224374, 'f1': 0.41001556791804356, 'acc': 0.4551246537396122, 'precision': 0.40066662047038093, 'recall': 0.506089566020314}
[rank0]: Traceback (most recent call last):
[rank0]:   File "/workspace/FlashRAG/examples/methods/run_exp.py", line 562, in <module>
[rank0]:     func(args)
[rank0]:   File "/workspace/FlashRAG/examples/methods/run_exp.py", line 337, in skr
[rank0]:     result = pipeline.run(test_data)
[rank0]:   File "/workspace/FlashRAG/flashrag/pipeline/pipeline.py", line 174, in run
[rank0]:     dataset = self.evaluate(dataset, do_eval=do_eval, pred_process_fun=pred_process_fun)
[rank0]:   File "/workspace/FlashRAG/flashrag/pipeline/pipeline.py", line 42, in evaluate
[rank0]:     self.retriever._save_cache()
[rank0]: AttributeError: 'NoneType' object has no attribute '_save_cache'
```

**Comment by ignorejjj** [comment]
I got it! Have fixed this bug in commit aeaf2e4bd4908dd6cd7d9d8cda0057380291d5ab.

**Comment by codecodebear** [comment]
Thank you for the fix! works now.
